### PR TITLE
rascaline was renamed to featomic

### PR DESCRIPTION
**Pull request by Luthaf**
This update the special handling for the corresponding TorchScript extension, and all references in the code/docs.

# Contributor (creator of pull-request) checklist

 - [x] Tests updated (for new features and bugfixes)?
 - [x] Documentation updated (for new features)?
 - [ ] ~Issue referenced (for PRs that solve an issue)?~

# Reviewer checklist

 - [ ] CHANGELOG updated with public API or any other important changes?

----
 📚 [Download documentation preview for this pull-request](https://nightly.link/metatensor/metatensor/actions/artifacts/2202079835.zip)

### CODE REVIEW

File: python/metatensor-torch/metatensor/torch/atomistic/_extensions.py [hunk 1]
```diff
@@ -12,15 +12,24 @@
...
+def _featomic_lib_path():
+    import featomic
+
+    return [featomic._c_lib._lib_path()]
+
+
 # Manual definition of which TorchScript extensions have their own dependencies. The
 # dependencies should be returned in the order they need to be loaded.
 EXTENSIONS_WITH_DEPENDENCIES = {
     "rascaline_torch": _rascaline_lib_path,
```
Comment (jwa7) [inline python/metatensor-torch/metatensor/torch/atomistic/_extensions.py:31]: Is this part of the backward compatibility too, or should it be removed?

**Review by jwa7 — COMMENTED** [review]
Looks good, but one question:

File: python/metatensor-torch/metatensor/torch/atomistic/_extensions.py (hunk 1 above)
Comment (Luthaf) [reply python/metatensor-torch/metatensor/torch/atomistic/_extensions.py:31]: Yes, the function above is only used here, so both should be kept & removed at the same time

Comment (jwa7) [reply python/metatensor-torch/metatensor/torch/atomistic/_extensions.py:31]: Ok great!

**Review by jwa7 — APPROVED** [review]

**Pull request merged**